libipl/p10/common.C

```diff
@@ -287,3 +290,44 @@ int ipl_set_sbe_state_all_sec(enum sbe_state state)
 	}
 	return ret;
 }
+
+void ipl_process_fapi_error(const fapi2::ReturnCode& fapirc,
+		struct pdbg_target *target, bool deconfig)
+{
+	if(fapirc == fapi2::FAPI2_RC_SUCCESS) {
+		ipl_error_callback(IPL_ERR_OK);
+	} 
+	else if(fapirc.getCreator() == fapi2::ReturnCode::CREATOR_PLAT) {
+		CDG_Target cdgTarget;
+		cdgTarget.callout_priority =
+		fapi2::plat_CalloutPriority_tostring(fapi2::CalloutPriorities::MEDIUM);
+
+		ATTR_PHYS_BIN_PATH_Type physBinPath;
+		uint32_t binPathElemCount = dtAttr::fapi2::ATTR_PHYS_BIN_PATH_ElementCount;
+		if (!pdbg_target_get_attribute(target, "ATTR_PHYS_BIN_PATH",
+			std::stoi(dtAttr::fapi2::ATTR_PHYS_BIN_PATH_Spec),
+				binPathElemCount, physBinPath)) {
+			ipl_log(IPL_ERROR, "Failed to read ATTR_PHYS_BIN_PATH for target %s\n",
+				pdbg_target_path(target));
+			ipl_error_callback(IPL_ERR_ATTR_READ_FAIL);
+		}
+		else {
+			std::copy(physBinPath, physBinPath+binPathElemCount,
+			std::back_inserter(cdgTarget.target_entity_path));
+			cdgTarget.deconfigure = deconfig;
+
+			// ffdc pointer will be deleted in the desturctor of 
+			FFDC ffdc;
+			ffdc.hwp_errorinfo.cdg_targets.push_back(cdgTarget);
+			ffdc.ffdc_type = FFDC_TYPE_HWP;
+			uint32_t rc = fapirc;
+			ffdc.hwp_errorinfo.rc = std::to_string(rc);
```

Comment:
are we get string value or numeric value? for HWP error we will get string representation (more human-readable format)

Reply:
fapirc is from ReturnCode class and the value we get is uint32_t, which I am converting to string